File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlNotExpression.cs
namespace T1.SqlSharp.Expressions;

public class SqlNotExpression : ISqlExpression
{
    public SqlType SqlType { get; set; } = SqlType.NotExpression;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_NotExpression(this);
    }

    public required ISqlExpression Value { get; set; }
    public string ToSql()
    {
        return $"NOT {Value.ToSql()}";
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlNotExpression.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlNotExpression implements ISqlExpression {
    SqlType: SqlType = SqlType.NotExpression;
    Span: TextSpan = new TextSpan();
    Value!: ISqlExpression;

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_NotExpression(this);
    }

    ToSql(): string {
        return `NOT ${this.Value.ToSql()}`;
    }
} 